bigc/resources/customers.py

```diff
@@ -81,6 +85,40 @@ def update_form_field(self, customer_id: int, field_name: str, value: Any) -> di
         }]
         return self._api.v3.put('/customers/form-field-values', json=payload)[0]
 
-    def delete(self, customer_id: int) -> None:
-        """Delete a specific customer by its ID"""
-        self._api.v3.delete(f'/customers?id:in={customer_id}')
+    def all_addresses(self, customer_id: int) -> Iterator[dict]:
+        """Get all addresses, filtering for a single customer"""
+        return self._api.v3.get_many(f'/customers/addresses?customer_id:in={customer_id}')
+
+    def get_address(self, customer_id: int, address_id: int) -> dict:
+        """Get one address by its ID, filtering for a single customer"""
+        return self._api.v3.get(f'/customers/addresses?customer_id:in={customer_id}&id:in={address_id}')
+
+    def create_address(self, address_data: dict) -> dict:
+        """Create an address"""
+        valid_address_params = ['customer_id', 'first_name', 'last_name', 'company', 'phone', 'address_type',
+                                'address1', 'address2', 'city', 'country_code', 'state_or_province', 'postal_code']
+
+        validated_address_data = [{
+            address_key: address_value
+            for address_key, address_value in address_data.items()
+            if address_key in valid_address_params
+        }]
+
+        return self._api.v3.post('/customers/addresses', json=validated_address_data)
+
+    def update_address(self, address_data: dict) -> dict:
+        """Update an address by its ID"""
+        valid_address_params = ['id', 'customer_id', 'first_name', 'last_name', 'company', 'phone', 'address_type',
+                                'address1', 'address2', 'city', 'country_code', 'state_or_province', 'postal_code']
+
+        validated_address_data = [{
+            address_key: address_value
+            for address_key, address_value in address_data.items()
+            if address_key in valid_address_params
+        }]
+
+        return self._api.v3.put('/customers/addresses', json=validated_address_data)
+
+    def delete_address(self, addresses_to_delete: list) -> dict:
+        """Delete an address by its ID"""
+        return self._api.v3.delete(f"/customers/addresses?id:in={','.join(str(i) for i in addresses_to_delete)}")
```

Comment:
This is still a bulk delete when it should just be deleting a single address.

```suggestion
    def delete_address(self, address_id: int) -> None:
        """Delete an address from a customer's address book"""
        self._api.v3.delete(f'/customers/addresses?id:in={address_id}')
```